pkg/testing/tools/kibana.go

```diff
@@ -0,0 +1,91 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package tools
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/elastic/elastic-agent-libs/kibana"
+)
+
+type DashboardResponse struct {
+	Page         int         `json:"page"`
+	PerPage      int         `json:"per_page"`
+	Total        int         `json:"total"`
+	SavedObjects []Dashboard `json:"saved_objects"`
+}
+
+type Dashboard struct {
+	Type       string    `json:"type"`
+	ID         string    `json:"id"`
+	Namespaces []string  `json:"namespaces"`
+	UpdatedAt  time.Time `json:"updated_at"`
+	CreatedAt  time.Time `json:"created_at"`
+	Version    string    `json:"version"`
+}
+
+// DeleteDashboard removes the selected dashboard
+func DeleteDashboard(ctx context.Context, client *kibana.Client, id string) error {
+	// In the future there should be logic to check if we need this header, waiting for https://github.com/elastic/kibana/pull/164850
+	headers := http.Header{}
+	headers.Add("x-elastic-internal-origin", "integration-tests")
+	status, resp, err := client.Connection.Request("DELETE", fmt.Sprintf("/api/saved_objects/dashboard/%s", id), nil, headers, nil)
+	if err != nil {
+		return fmt.Errorf("error making API request: %w, response: '%s'", err, string(resp))
+	}
+
+	if status != 200 {
```

Comment:
A little suggestion:
```suggestion
	if status != http.StatusOK {
```